stagecoach/price-pusher/price_pusher/core/poller.py

```diff
@@ -0,0 +1,68 @@
+import logging
+
+from abc import ABC, abstractmethod
+from typing import List, Dict, Optional, Callable
+
+from pragma.core.entry import Entry
+from pragma.publisher.client import FetcherClient
+
+from price_pusher.utils.retries import retry_async
+
+logger = logging.getLogger(__name__)
+
+FnUpdatePrices = Callable[[List[Entry]], None]
+
+
+class IPricePoller(ABC):
+    fetcher_client: FetcherClient
+    update_prices_callback: Optional[FnUpdatePrices]
+
+    @abstractmethod
+    def set_update_prices_callback(self, callback: FnUpdatePrices) -> None: ...
+
+    @abstractmethod
+    async def poll_prices(self) -> None: ...
+
+
+class PricePoller(IPricePoller):
+    def __init__(self, fetcher_client: FetcherClient) -> Dict:
+        self.fetcher_client = fetcher_client
+        self.update_prices_callback = None
+
+    def is_requesting_onchain(self) -> bool:
```

Comment:
add doc

Reply:
removed this function & replaced by one property with docs